### PR TITLE
WIP: fix ZipSlip problem in cpio package

### PR DESCRIPTION
**Pull request by rminnich**

### CODE REVIEW

**Comment by rminnich** [comment]
I think the ZipSlip test belongs in u-root utils, right? This is just a simple pass, and we need a set of test cases for ZipSlip.

File: pkg/cpio/utils.go [hunk 1]
```diff
@@ -256,3 +256,15 @@ func Equal(r Record, s Record) bool {
...
 }
+
+func ZipSlip(dest, archiveName string) error {
+	clean := filepath.Clean(archiveName)
+	relDest, err := filepath.Rel(dest, clean)
+	if err != nil {
+		return fmt.Errorf("ZipSlip test can not form a path from %q, %q", dest, clean)
+	}
+	if strings.HasPrefix(relDest, "..") {
+		return fmt.Errorf("Error: rel path %q has .. as the prefix and hence fails the ZipSlip test", relDest)
```
Comment (hugelgupf) [inline pkg/cpio/utils.go:267]: can you explain here why it's being rejected, rather than mentioning ZipSlip? I'm pretty sure users won't know what that is

Comment (rminnich) [reply pkg/cpio/utils.go:267]: I don't like this anyway. It's much simpler just to do the path creation correctly, or at least I think it is correct, wdyt, to do:
var base, file string
newpath := filepath.Join(base, filepath.Clean(filepath.Join(os.PathSeparator, file)))

When the file has the / put at front via the Join, the Clean swallows up all the .. in the file name, such that when joined to the base, ZipSlip can't happen. Then all this silliness is not needed.

**Comment by rminnich** [comment]
this is the wrong way to do it.

I am going to start a discussion in the go repo about this.

**Pull request closed without merging**